x-pack/plugins/observability_solution/infra/server/lib/helpers/get_infra_metrics_client.test.ts

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KibanaRequest } from '@kbn/core-http-server';
+import { DataTier } from '@kbn/observability-shared-plugin/common';
+import { InfraBackendLibs } from '../infra_types';
+import { getInfraMetricsClient } from './get_infra_metrics_client';
+import { InfraPluginRequestHandlerContext } from '../../types';
+import { QueryDslQueryContainer } from '@kbn/data-views-plugin/common/types';
+
+const withExcludedDataTiers = (tiers: DataTier[]) => ({
+  uiSettings: {
+    client: {
+      get: () => Promise.resolve(tiers),
+    },
+  },
+});
+
+const mockedInfra = { getMetricsIndices: () => Promise.resolve(['*.indices']) };
+
+const infraMetricsTestHarness =
+  (input: QueryDslQueryContainer | undefined, output: QueryDslQueryContainer) => async () => {
+    const callWithRequest = jest.fn();
+
+    const mockedCore = withExcludedDataTiers(['data_frozen']);
+
+    const context = {
+      infra: Promise.resolve(mockedInfra),
+      core: Promise.resolve(mockedCore),
+    } as unknown as InfraPluginRequestHandlerContext;
+
+    const client = await getInfraMetricsClient({
+      libs: { framework: { callWithRequest } } as unknown as InfraBackendLibs,
+      context,
+      request: {} as unknown as KibanaRequest,
+    });
+
+    await client.search({
+      body: {
+        query: input,
+        size: 1,
+        track_total_hits: false,
+      },
+    });
+
+    expect(callWithRequest).toBeCalledWith(
+      context,
+      'search',
+      {
+        body: {
+          query: output,
+          size: 1,
+          track_total_hits: false,
+        },
+        ignore_unavailable: true,
+        index: ['*.indices'],
+      },
+      {}
+    );
+  };
+
+describe('getInfraMetricsClient', () => {
+  it(
+    'includes excluded data tiers in the request filter by default',
+    infraMetricsTestHarness(undefined, {
+      bool: {
+        filter: [
+          {
+            bool: {
+              must_not: [
+                {
+                  terms: {
+                    _tier: ['data_frozen'],
+                  },
+                },
+              ],
+            },
+          },
+        ],
+      },
+    })
+  );
+
+  it(
+    'merges provided filters with the excluded data tier filter',
+    infraMetricsTestHarness(
+      {
+        bool: {
+          filter: [
+            {
+              bool: {
+                must: {
+                  exists: {
+                    field: 'a-field',
+                  },
+                },
+              },
+            },
+          ],
+        },
+      },
+      {
```

Comment:
This is not correct. It's possible to have many `bool` objects inside the `filter` array, but there can't be multiple `bool` in the `query` object. Running this on devtools returns the following error

```
{
  "error": {
    "root_cause": [
      {
        "type": "parsing_exception",
        "reason": "Unknown key for a START_OBJECT in [bool].",
        "line": 1,
        "col": 12
      }
    ],
    "type": "parsing_exception",
    "reason": "Unknown key for a START_OBJECT in [bool].",
    "line": 1,
    "col": 12
  },
  "status": 400
}
```